crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -596,24 +596,39 @@ impl Room {
                 let name = name.trim();
                 return Ok(DisplayName::Named(name.to_owned()));
             }
+
             if let Some(alias) = inner.canonical_alias() {
                 let alias = alias.alias().trim();
                 return Ok(DisplayName::Aliased(alias.to_owned()));
             }
+
             inner.summary.clone()
         };
 
         let own_user_id = self.own_user_id().as_str();
 
-        let members: Vec<RoomMember> = if summary.heroes.is_empty() {
-            self.members(RoomMemberships::ACTIVE)
-                .await?
-                .into_iter()
-                .filter(|u| u.user_id() != own_user_id)
-                .take(5)
-                .collect()
+        let (heroes, guessed_num_members): (Vec<RoomMember>, _) = if summary.heroes.is_empty() {
+            let mut members = self.members(RoomMemberships::ACTIVE).await?;
+
+            // Make the ordering deterministic.
+            members.sort_unstable_by(|lhs, rhs| lhs.name().cmp(rhs.name()));
+
+            // We can make a good prediction of the total number of members here. This might
+            // be incorrect if the database info is outdated.
+            let guessed_num_members = Some(members.len());
+
+            // The server returns at most 5 heroes; behave similarly here.
```

Comment:
Do we have a reference to it, a proof, something 🙂?

Reply:
Are you comfortable by quoting (a part of) the spec with a link to it? We already do that in the code when it needs specific clarifications.

Reply:
Sure, but the *should* isn't that confidence inspiring.

Reply:
Indeed… at least we can consider it's a safe number.